src/OrbitGl/App.cpp

```diff
@@ -349,6 +342,16 @@ void OrbitApp::OnTracepointEvent(orbit_client_protos::TracepointEventInfo tracep
       is_same_pid_as_target);
 }
 
+void OrbitApp::OnModuleUpdate(uint64_t /*timestamp_ns*/, ModuleInfo module_info) {
+  module_manager_->AddOrUpdateModules({module_info});
+  GetMutableCaptureData().mutable_process()->AddOrUpdateModuleInfo(module_info);
+}
+
+void OrbitApp::OnModulesSnapshot(uint64_t /*timestamp_ns*/, std::vector<ModuleInfo> module_infos) {
+  module_manager_->AddOrUpdateModules(module_infos);
```

Comment:
This call returns a `std::vector<ModuleData*>` of modules that are not loaded (do not contain symbols) anymore after this call. I think the `AddOrUpdateModules` method should be `[[nodiscard]]`, because this rare case needs to be taken care of. After this call, the functions data view might contain pointers to FunctionInfos that do not exist anymore. 

There is the method `OrbitApp::ReloadModules` which is the only place `AddOrUpdateModules` was called before. `ReloadModules` attempts to load the symbols again for modules unloaded, which should not be done here. So probably the best way forward here is to split `ReloadModules` into 2 parts, the first that does the proper unloading (including refreshing the dataviews) and a second part that loads symbols again. 

@beckerhe touched this `ReloadModules` most recently, and is planning to move symbol loading related code into a separate module.

Reply:
Looking at the LiveFunctions it seems like this should result in one of following for hooked functions:
1. Service will (probably) stop sending events if module has been changed and was reloaded by the process. In which case we it does not matter whether or not we refresh the view.
2. Service will keep sending events because even though module was updated and loaded the old image is still there and the executable might still be calling it. In which case we want live view to have old functions in the list. Note that Live Function view keeps copies of FunctionInfos.

I will check again there could be problems with that but from the first look it seems like this is exactly what we want. we do not want to rehook the functions, because this will result in undefined behavior for the second case.

Reply:
after offline discussion - we still have a problem here with FunctionDataView because it uses pointers to FuctionInfos. And since in the case we have too many things that could go wrong or result in incorrect data in callstacks/sampling reports when a module without buildid gets reloaded - the decision here is to generate a error and stop/abort.